src/internal/connector/graph/api/api_test.go

```diff
@@ -0,0 +1,114 @@
+package api_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/connector/graph/api"
+)
+
+type mockNextLink struct {
+	nextLink *string
+}
+
+func (l mockNextLink) GetOdataNextLink() *string {
+	return l.nextLink
+}
+
+type mockDeltaNextLink struct {
+	mockNextLink
+	deltaLink *string
+}
+
+func (l mockDeltaNextLink) GetOdataDeltaLink() *string {
+	return l.deltaLink
+}
+
+type testInput struct {
+	name         string
+	inputLink    *string
+	expectedLink string
+}
+
+// Needs to be var not const so we can take the address of it.
+var (
+	emptyLink = ""
+	link      = "foo"
+	link2     = "bar"
+
+	basicInputs = []testInput{
```

Comment:
Nit: seems like this is the expected usage of this table.
```suggestion
	nextLinkInputs = []testInput{
```